doc/development.rst

```diff
@@ -172,3 +172,186 @@ If there are open questions, final decisions are made by the Temporary
+
+In addition, we can have a look at `MANIFEST.in` to check that all additional files that we want to be included or excluded from the release are indicated. Normally we shouldn't have to touch this file.
+
+Add these changes and the version tag, and commit them:
+
+.. code:: bash
+    
+    git add doc/whats_new.rst nilearn.version.py
+    git commit -m "REL x.y.z"
+    git tag x.y.z
+    git push origin release-x.y.z
+    git push --tags
+
+
+Build the distributions and upload them to Pypi
+-----------------------------------------------
+
+First of all we should make sure we don't include files that shouldn't be present:
+
+.. code-block:: bash
+    
+    git checkout x.y.z
+
+
+If the workspace contains a `dist` folder, make sure to clean it:
+
+.. code-block:: bash
+    
+    rm -r dist
+
+
+In order to build the binary wheel files, we need to install `wheel <https://pypi.org/project/wheel/>`_:
+
+.. code-block:: bash
+    
+    pip install wheel
+
+
+And, in order to upload to `Pypi`, we will use `twine <https://pypi.org/project/twine/>`_ that you can also install with `pip`:
+
+.. code-block:: bash
+    
+    pip install twine
+
+
+Build the source and binary distributions:
+
+.. code-block:: bash
+    
+    python setup.py sdist bdist_wheel
+
+
+This should add two files to the `dist` subfolder:
+
+- one for the source distribution that should look like `PACKAGENAME-VERSION.tar.gz`
+- one for the binary distribution that should look like `PACKAGENAME-PACKAGEVERSION-PYTHONVERSION-PYTHONCVERSION-PLATFORM.whl`
+  
+Optionally, we can run some basic checks with `twine`:
+
+.. code-block:: bash
+    
+    twine check dist/*
+
+
+We are now ready to upload to `Pypi`. Note that you will need to have an account on Pypi (you can create one `here <https://pypi.org/account/register/>`_), and be added to the maintainers of `Nilearn <https://pypi.org/project/nilearn/>`_. If you satisfy these conditions, you should be able to run:
+
+.. code-block:: bash
+    
+    twine upload dist/*
+
+
+Once the upload is completed, make sure everything looks good on `Pypi <https://pypi.org/project/nilearn/>`_. Otherwise you will probably have to fix the issue and start over a new release with the patch number incremented.
+
+At this point, we need to upload the binaries to GitHub and link them to the tag. To do so, go to the `Nilearn GitHub page <https://github.com/nilearn/nilearn/tags>`_ under the "Releases" tab, and edit the `x.y.z` tag by providing a description, and upload the distributions we just created (you can just drag and drop the files).
+
+
+Build and deploy the documentation
+----------------------------------
+
+We now need to update the documentation:
+
+.. code-block:: bash
+    
+    cd doc
+    make install
+
+
+This will build the documentation (beware, this is time consuming...) and push it to the `GitHub pages repo <https://github.com/nilearn/nilearn.github.io>`_.
```

Comment:
Maybe you could give here the push command you use (not completely obvious from the context).

Reply:
Although I never ran this before, I don't think you have to do anything more than that. All the git commands are in the `install` target:

https://github.com/nilearn/nilearn/blob/80572dbf2253bd7a5a00e688082cbd3075e71a12/doc/Makefile#L161-L178

Reply:
OK, I misread it. thx.